### PR TITLE
BREAKING CHANGE: Refactor inputoptions component

### LINKED ISSUE #4: Break up complexity of a single ods-inputoptions component
# Issue report

Your issue may already be reported! Please search on the issue tracker before creating one.

## Expected Behavior
1. Input radio and checkboxes are separate components
1. A wrapper component that allows only a single radio button to be selected at once
1. The visible design and interactions remain the same as the current `ods-inputoptions`

## Current Behavior
1. The single `ods-inputoptions` is a fairly complex component and exposes properties that compose multiple HTML input properties.

## Possible Solution
1. Break out `checkbox` and `radio` with label into separate components. Radio elements should be groupable.
1. Create wrapper component that allows access to HTML input properties.

## Context
1. This change should allow input options to be a controlled component from React (i.e. allow the outer application to manage the state of the input radio/checkboxes)

## References
1. [@polymer/paper-radio-group](https://www.webcomponents.org/element/@polymer/paper-radio-group)
1. [@polymer/paper-radio-button](https://www.webcomponents.org/element/@polymer/paper-radio-button)
1. [@polymer/paper-checkbox](https://www.webcomponents.org/element/@polymer/paper-checkbox)
1. [input/radio](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/radio)

### LINKED ISSUE #9: onInput event firing twice
# Issue report
## Expected Behavior
- onInput event should fire once
- onChange does not fire an event

## Current Behavior
- onInput event fires twice
- onChange should fire an event

## Possible Solution
- Talked with Geoff, and he mentioned that he could have something to do with how ods-inputoptions is surfacing the events 

## Steps to Reproduce (for bugs)
https://github.com/AlaskaAirlines/OrionReactDemo/blob/master/src/Components.js#L62
- Link above was to the demo site github. Should be able to clone and run the site locally with a custom console log onInput, and you should see two events fired twice.

## Context
- Trying to omniture tag how our users are changing their mind when preordering

## Your Environment
* Browser Name and version: Chrome Version 78.0.3904.108 (Official Build) (64-bit)
* Operating System and version (desktop or mobile): Desktop Windows

### LINKED ISSUE #10: Update Orion to Auro
# User story

As a user, I want to be able to use the latest and greatest UI, so I can be awesome with Auro.

## Outline subtasks or tasks

> Decide which specific steps need to be completed and who is responsible for each of them.

- [x] remove Orion UI
- [x] remove the dependency on shape.json (optional)
- [ ] update UI of radio buttons and checkboxes to fit Auro UI spec
- [x] update `"name": "@alaskaairux/ods-inputoptions",` to `"name": "@alaskaairux/auro-inputoptions",`
- [ ] update Auro Doc site with new component 

Once the PR is merged and all work completed: 
- [ ] update repo name to be `AlaskaAirlines/auro-inputoptions`

## Definition of “Done” (exit criteria)

> The story is generally “done” when the user can complete the outlined task, but make sure to define what that is.

This story will be considered complete once the radio button and checkbox elements fit the new Auro design spec, and the next major release has a new npm name and the doc site is updated with the new content.

### PR DESCRIPTION
**Pull request by blackfalcon**
# Alaska Airlines Pull Request

Update currently released ods-inputoptions with two new components to reduce the complexity of support. 

**Fixes:** #4, #9, #10

## Summary:

The core features of ods-inputoptions will be preserved and with this update will release two new components for use. ods-inputoption-group and ods-inputoption.

ods-inputoption-group will be a wrapper component used within the HTML and ods-inputoption cab be used repeatedly as elements within the slot of ods-inputoption-group. This will allow for a less complex API of the component as well as give developers direct access to the inner components. 

Worked within the scope of this update also includes support for a light DOM version of the component. 

## Type of change:

Please delete options that are not relevant.

- [x] New capability 
- [x] Revision of an existing capability
- [ ] Infrastructure change (automation, etc.)
- [ ] Other (please elaborate) 

## Checklist:

- [x] My update follows the CONTRIBUTING guidelines of this project
- [x] I have performed a self-review of my own update

**By submitting this Pull Request, I confirm that my contribution is made under the terms of the Apache 2.0 license.**

_Pull Requests will be evaluated by their quality of update and whether it is consistent with the goals and values of this project. Any submission is to be considered a conversation between the submitter and the maintainers of this project and may require changes to your submission._ 

**Thank you for your submission!**<br>
-- Orion Design System Team

### CODE REVIEW

File: src/ods-inputoption-group.js [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+
+      ${this._errorChange()}
+
+      ${this.label ?
+        html`<label for="${this.for}" class="ods-label">${this.label}</label>` :
+        html``}
+
+      <div
+        @input="${this._handleInput}"
+        class="${classMap(groupClasses)}">
```
Comment (blackfalcon) [inline src/ods-inputoption-group.js]: I really like this. Just started looking at classMap too and seeing how other components can be refactored.

**Review by AuroDesignSystem — CHANGES_REQUESTED** [review]
Overall, looking great. But there is an issue with the a11y tabbing states. The user will tab INTO a radio button set and use the ARROW KEYS to move between buttons. 

Then the user will tab OUT of the set. 

The way this is currently working, the a11y user is trapped.  Please review the a11y interaction on the MASTER branch. 

There is the same issue with checkboxes. 

Tabbing appears to be totally broken with light DOM versions?

File: demo/index.html [hunk 2]
```diff
@@ -13,6 +13,9 @@
       import '@polymer/iron-demo-helpers/demo-snippet';
     </script>
     <script type="module" src="../src/ods-inputoptions.js"></script>
```
Comment (AuroDesignSystem) [inline demo/index.html]: ods-inputoptions should be removed from this version. This is a BREAKING CHANGE and there is no need to support this version any longer.

**Review by AuroDesignSystem — CHANGES_REQUESTED** [review]

**Review by AuroDesignSystem — CHANGES_REQUESTED** [review]
There seems to be a general issue with the tabbing. In the examples on the demo page, when I click on the LABEL, the first option is not selected? 

Please see https://auro-prod-site-w2.azurewebsites.net/ods-options for the initial version of this component. 

There seems to be some wonkyness with the CSS selector.

**Review by AuroDesignSystem — DISMISSED** [review]

**Review by AuroDesignSystem — APPROVED** [review]

**Pull request merged**